src/box/index.cc

```diff
@@ -142,27 +142,35 @@ key_validate(const struct index_def *index_def, enum iterator_type type,
 }
 
 int
-exact_key_validate(struct key_def *key_def, const char *key,
+exact_key_validate(struct index_def *index_def, const char *key,
 		   uint32_t part_count)
 {
 	assert(key != NULL || part_count == 0);
+	struct key_def *key_def = index_def->key_def;
 	if (key_def->part_count != part_count) {
-		diag_set(ClientError, ER_EXACT_MATCH, key_def->part_count,
-			 part_count);
+		struct space *space = space_by_id(index_def->space_id);
+		diag_set(ClientError, ER_EXACT_MATCH,
+			 tt_sprintf("in space \"%s\", index \"%s\" ",
+				    space_name(space), index_def->name),
+			 key_def->part_count, part_count);
```

Comment:
1. Updated the submodule in [Tarantool EE](https://github.com/tarantool/tarantool-ee) to this PR.

2. Applied the following patch to make the project compile.

   <details>

   ```diff
      diff --git a/src/box/index_read_view.c b/src/box/index_read_view.c
   index 7c6c562fef9a..d4dfa56b29f4 100644
   --- a/src/box/index_read_view.c
   +++ b/src/box/index_read_view.c
   @@ -190,7 +190,7 @@ box_index_read_view_get(struct index_read_view *rv, const char *key,
    		return -1;
    	}
    	uint32_t part_count = mp_decode_array(&key);
   -	if (exact_key_validate(rv->def->key_def, key, part_count) != 0)
   +	if (exact_key_validate(rv->def, key, part_count) != 0)
    		return -1;
    	if (index_read_view_get(rv, key, part_count, result) != 0)
    		return -1;
   diff --git a/src/box/raw_read_view.c b/src/box/raw_read_view.c
   index 4e83bbd8850f..a0038bf464a4 100644
   --- a/src/box/raw_read_view.c
   +++ b/src/box/raw_read_view.c
   @@ -318,7 +318,7 @@ box_raw_read_view_get(const box_raw_read_view_index_t *index_,
    		return -1;
    	}
    	uint32_t part_count = mp_decode_array(&key);
   -	if (exact_key_validate(index->def->key_def, key, part_count) != 0)
   +	if (exact_key_validate(index->def, key, part_count) != 0)
    		return -1;
    	return index_read_view_get_raw(index, key, part_count, data, size);
    }
   ```

   </details>

3. Ran the following script.

   <details>

   ```Lua
   box.cfg({log_level = 'warn'})
   box.schema.create_space('test')
   box.space.test:create_index('pk')
   rv = box.read_view.open()
   box.space.test:drop()
   rv.space.test:get({1, 2})
   ```

   </details>

4. Got a segmentation fault.

   <details>

   ```
   Segmentation fault
     code: SEGV_MAPERR
     addr: 0xc0
     context: 0x7facfedff540
     siginfo: 0x7facfedff670
     rax      0x0                0
     rbx      0x55ff7b7240a0     94555776106656
     rcx      0x14               20
     rdx      0x100000           1048576
     rsi      0x100000           1048576
     rdi      0x0                0
     rsp      0x7facfedffc00     140380987194368
     rbp      0x7facfedffc00     140380987194368
     r8       0x7facfedfffd0     140380987195344
     r9       0x0                0
     r10      0x7facff03b000     140380989534208
     r11      0x10               16
     r12      0x1                1
     r13      0x0                0
     r14      0x40dfc088         1088405640
     r15      0x55ff7a7bd4b0     94555759957168
     rip      0x55ff7a57bd07     94555757591815
     eflags   0x10246            66118
     cs       0x33               51
     gs       0x0                0
     fs       0x0                0
     cr2      0xc0               192
     err      0x4                4
     oldmask  0x0                0
     trapno   0xe                14
   Current time: 1676469433
   Please file a bug at https://github.com/tarantool/tarantool/issues
   Attempting backtrace... Note: since the server has already crashed,
   this may fail as well
   #1  0x55ff7a7f42d6 in crash_collect+256
   #2  0x55ff7a7f4ce8 in crash_signal_cb+100
   #3  0x7fad016da420 in __restore_rt+0
   #4  0x55ff7a57bd07 in space_name+12
   #5  0x55ff7a57cdc5 in exact_key_validate+257
   #6  0x55ff7a7b8365 in box_index_read_view_get+588
   #7  0x55ff7a7bd591 in lbox_index_read_view_get+225
   #8  0x55ff7a8589f7 in lj_BC_FUNCC+70
   #9  0x55ff7a863875 in lua_pcall+849
   #10 0x55ff7a7d2f78 in luaT_call+45
   #11 0x55ff7a7c8c58 in lua_main+189
   #12 0x55ff7a7c9200 in run_script_f+1423
   #13 0x55ff7a565cca in fiber_cxx_invoke(int (*)(__va_list_tag*), __va_list_tag*)+34
   #14 0x55ff7a80016d in fiber_loop+210
   #15 0x55ff7aaca17d in coro_init+120
   Aborted
   ```

   </details>